modules/ding_place2book/ding_place2book.module

```diff
@@ -571,36 +574,63 @@ function ding_place2book_get_event_state($event_maker_id, $event_id) {
     if (!empty($ticket_link_status)) {
       switch ($ticket_link_status) {
         case 'open':
-          $data = l(t('Book a ticket'), $event->links->sales_location, array('attributes' => array('target' => '_blank')));
           $class = 'ticket-available';
+          $data = l(t('Book a ticket'), $event->links->sales_location, array(
+            'attributes' => array(
+              'target' => '_blank',
+              'class' => array('button', $class),
+            ),
+          ));
+          $type = 'link';
           break;
 
         case 'closed':
-          $data = t('Not open for ticket sale');
           $class = 'ticket-notice';
+          $data = t('Not open for ticket sale');
+          $type = 'markup';
           break;
 
         case 'sold-out':
-          $data = t('Sold out');
           $class = 'ticket-warning';
+          $data = t('Sold out');
+          $type = 'markup';
           break;
 
         case 'upcoming':
-          $data = t('Ticket sale has not yet started for this event');
+          $sale_start = $price->sale_begin_at;
```

Comment:
So where does this $price variable come from? Is it the last assignment to $price in the foreach further up? Are we even sure this is the next open sales period?

I think we need to remember the closest next sales period, when we find an upcoming sales period when iterating the sales periods.